numba/cuda/tests/cudadrv/test_cuda_array_slicing.py

```diff
@@ -97,6 +97,18 @@ def test_prefix_select(self):
         darr = cuda.to_device(arr)
         self.assertTrue(np.all(darr[:1, 1].copy_to_host() == arr[:1, 1]))
 
+    def test_empty_slice_1d(self):
+        arr = np.arange(5)
+        darr = cuda.to_device(arr)
+        for i in range(darr.shape[0]):
+            np.testing.assert_array_equal(darr[i:i].copy_to_host(), arr[i:i])
+
+    def test_empty_slice_2d(self):
+        arr = np.arange(5 * 5).reshape(5, 5)
+        darr = cuda.to_device(arr)
+        np.testing.assert_array_equal(darr[:0].copy_to_host(), arr[:0])
+        np.testing.assert_array_equal(darr[3, :0].copy_to_host(), arr[3, :0])
+
```

Comment:
Instead of raising, numpy just return an empty array.  In https://github.com/numba/numba/pull/2434/commits/282beeee81c637d5339ccb3df74e6b9ad93b5f72, tests are added to ensure that.